src/options/options.js

```diff
@@ -1,3 +1,191 @@
+    }
+  });
+}
+
+/*
+On startup, check whether we have stored settings.
+If we don't, then store the default settings.
+*/
+async function checkStoredSettings(storedSettings) {
+  if (!storedSettings.storedShortCutsArr) {
+    browser.storage.local.set({ storedShortCutsArr: defaultShortCutsArr });
+  }
+
+  const settings = await browser.storage.local.get('storedShortCutsArr');
+
+  await setShortCutsOnLoadPage(settings);
+}
+
+browser.storage.local.get().then(checkStoredSettings, onError);
+
+sidebarMenuTabs.forEach((item) => {
+  item.addEventListener('click', (e) => {
+    if (e.target.checked) {
+      optionsTitle.innerText = e.target.value;
+    }
+  });
+});
+
+// Clear button handler
+shortcutClearBtnArray.forEach((item) => {
+  const shortcutClearBtnElement = item;
+  shortcutClearBtnElement.addEventListener('click', () => {
+    const currentInput = shortcutClearBtnElement.previousElementSibling;
+    currentInput.value = '';
+    browser.commands.update({ name: currentInput.id, shortcut: '' });
+    browser.storage.local.set({ storedShortCutsArr: [{ id: currentInput.id, shortcut: '' }] });
+    shortcutClearBtnElement.parentNode.nextElementSibling.innerText = '';
+  });
+});
+
+// Reset button handler
+shortcutResetBtnArray.forEach((item) => {
+  const shortcutResetBtnElement = item;
+  shortcutResetBtnElement.addEventListener('click', () => {
+    const currentInput = shortcutResetBtnElement.parentNode.firstChild.nextSibling;
+    const getDefaultShortCutsById = defaultShortCutsArr.find((el) => el.id === currentInput.id);
+    currentInput.value = getDefaultShortCutsById.shortcut;
+    browser.commands.update({ name: currentInput.id, shortcut: currentInput.value });
+    browser.storage.local.set({ storedShortCutsArr: [{ id: currentInput.id, shortcut: currentInput.value }] });
+    shortcutResetBtnElement.parentNode.nextElementSibling.innerText = '';
+  });
+});
+
+function getOS() {
+  const { userAgent } = window.navigator;
+  const platform = window.navigator?.userAgentData?.platform || window.navigator.platform;
+  const macosPlatforms = ['Macintosh', 'MacIntel', 'MacPPC', 'Mac68K'];
+  const windowsPlatforms = ['Win32', 'Win64', 'Windows', 'WinCE'];
+  const iosPlatforms = ['iPhone', 'iPad', 'iPod'];
+
+  if (macosPlatforms.indexOf(platform) !== -1) {
+    return 'Mac OS';
+  } if (iosPlatforms.indexOf(platform) !== -1) {
+    return 'iOS';
+  } if (windowsPlatforms.indexOf(platform) !== -1) {
+    return 'Windows';
+  } if (/Android/.test(userAgent)) {
+    return 'Android';
+  } if (/Linux/.test(platform)) {
+    return 'Linux';
+  }
+
+  return null;
+}
+const isMac = getOS() === 'Mac OS';
+
+const normalizeKey = (key, keyCode) => {
+  const alphabet = /^([a-z]|[A-Z])$/;
+  if (alphabet.test(key)) return key.toUpperCase();
+
+  const digit = /^[0-9]$/;
+  const func = /^F([0-9]|1[0-2])$/;
+  const homes = /^(Home|End|PageUp|PageDown|Insert|Delete)$/;
+  if (digit.test(key) || func.test(key) || homes.test(key)) return key;
```

Comment:
Analyze code from begin of function till this line.

For me it is obvious that `alphabet`, `digit` and `func` should result to the same `return`:
```
return key.toUpperCase();
```

But `homes` to different.

So grouping of conditions should be different. Isn't?